sphinx_gallery/gen_rst.py

```diff
@@ -1000,12 +1000,21 @@ def execute_code_block(
     sys.path.append(os.getcwd())
 
     # Save figures unless there is a `sphinx_gallery_defer_figures` flag
-    match = re.search(
+    defer_figs_match = re.search(
         r"^[\ \t]*#\s*sphinx_gallery_defer_figures[\ \t]*\n?",
         block.content,
         re.MULTILINE,
     )
-    need_save_figures = match is None
+    need_save_figures = defer_figs_match is None
+
+    # Add `sphinx_gallery_multi_image_block` setting to block variables
+    # (extract config rather than just regex search since the option's value is needed)
+    script_vars["multi_image"] = py_source_parser.extract_file_config(
+        block.content
+    ).get("multi_image_block")
```

Comment:
Would be interested if others think there's a better way to handle this.

Of course just a regex search is possible, but unlike the `defer_figures` flag we need to extract the value of variable, which the `extract_file_config()` func can handle.